packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AbstractModal from 'terra-abstract-modal';
+import Button from 'terra-button';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './NotificationDialog.scss';
+
+const cx = classNames.bind(styles);
+
+const variants = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  INFO: 'info',
+  SUCCESS: 'success',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * Header of the notification-dialog.
+   */
+  header: PropTypes.string,
+  /**
+   * Title of the notification-dialog.
+   */
+  title: PropTypes.string,
+  /**
+   * Message of the notification-dialog.
+   */
+  message: PropTypes.string,
+  /**
+   * The Action of the primary button.
+   */
+  primaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The Action of the secondary button.
+   */
+  secondaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The variant of notification to be rendered.
+   * Use one of alert, error, warning, info, success, custom.
+   */
+  variant: PropTypes.oneOf([
+    variants.ALERT,
+    variants.ERROR,
+    variants.WARNING,
+    variants.INFO,
+    variants.SUCCESS,
+    variants.CUSTOM,
+  ]),
+  /**
+   * The icon to be used for a notification of type custom. This will not be used for any other notification types.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * Toggle to show notification-dialog or not.
+   */
+  isOpen: PropTypes.bool.isRequired,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
+  title: null,
+  message: null,
+  variant: null,
+};
+
+const actionSection = (primaryAction, secondaryAction) => {
+  let actionButton = null;
+  let dismissButton = null;
+  if (!primaryAction && !secondaryAction) {
+    return null;
+  }
+  if (primaryAction) {
```

Comment:
If a primaryAction is not provided but a secondaryAction is, would we want render the secondary action with the Emphasis style?

Reply:
I think making primaryAction as required would make more sense. Thoughts?